src/Symfony/Component/Serializer/Context/ContextBuilderTrait.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Context;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+trait ContextBuilderTrait
+{
+    /**
+     * @var array<string, mixed>
+     */
+    protected array $context;
+
+    public function __construct(array $context = [])
```

Comment:
This could use constructor promotion, but I'm not sure that it's a good idea to include a constructor in a trait.

Reply:
Yes, I removed it. Indeed the end-user can manipulate `$context` directly instead

Reply:
But maybe, for that, they need a `withContext(array $context)` method, which was the purpose of the constructor